src/click/testing.py

```diff
@@ -348,7 +345,9 @@ def isolation(
         @_pause_echo(echo_input)  # type: ignore
         def visible_input(prompt: str | None = None) -> str:
             sys.stdout.write(prompt or "")
-            val = next(text_input).rstrip("\r\n")
+            if text_input_size and text_input.tell() == text_input_size:
+                raise EOFError()
+            val = text_input.readline().rstrip("\r\n")
```

Comment:
I would suggest simply moving the logic in `_NamedTextIOWrapper.__next__` to here:
```
try:
    val = next(text_input).rstrip("\r\n")
except StopIteration as e:
    raise EOFError() from e
```

This will do the same without having to check the input size. When I changed the relevant test, I thought that raising `Abort` at the end of stdin made sense since this is the behavior if the user uses Ctrl+C to end input but it seems that input can be ended cleanly with Ctrl+D.

Reply:
Simpler indeed, I have updated that PR accordingly.